app/views/editor/level/modals/SaveBranchModal.coffee

```diff
@@ -0,0 +1,185 @@
+ModalView = require 'views/core/ModalView'
+template = require 'templates/editor/level/modal/save-branch-modal'
+DeltaView = require 'views/editor/DeltaView'
+deltasLib = require 'core/deltas'
+Branch = require 'models/Branch'
+Branches = require 'collections/Branches'
+LevelComponents = require 'collections/LevelComponents'
+LevelSystems = require 'collections/LevelSystems'
+
+
+module.exports = class SaveBranchModal extends ModalView
+  id: 'save-branch-modal'
+  template: template
+  modalWidthPercent: 99
+  events:
+    'click #save-branch-btn': 'onClickSaveBranchButton'
+    'click #branches-list-group .list-group-item': 'onClickBranch'
+    'click .delete-branch-btn': 'onClickDeleteBranchButton'
+    'click #stash-branch-btn': 'onClickStashBranchButton'
+
+  initialize: (options = {}) ->
+    # Should be given all loaded, up to date systems and components with existing changes
+    { @components, @systems } = options
+    
+    # Create a list of components and systems we'll be saving a branch for
+    @componentsWithChanges = new LevelComponents(@components.filter((c) -> c.hasLocalChanges()))
+    @systemsWithChanges = new LevelSystems(@systems.filter((c) -> c.hasLocalChanges()))
+    
+    # Load existing branches
+    @branches = new Branches()
+    @branches.fetch({url: '/db/branches'})
+    .then(=>
+      
+      # Load any patch target we don't already have
+      fetches = []
+      for branch in @branches.models
+        for patch in branch.get('patches') or []
+          collection = if patch.target.collection is 'level_component' then @components else @systems
+          model = collection.get(patch.target.id)
+          if not model
+            model = new collection.model({ _id: patch.target.id })
```

Comment:
Ideally object constructors have capitalized names, not sure how much legacy code there is in this though. It's a bit confusing to me what is going on with "patch.target.collection.model" being a constructor.